### PR TITLE
[Feat] text align style

### PR DESCRIPTION
**Pull request by 5anniversary**
- Fixed a bug where size could get stuck at 0 due to not accepting decimal values while interpreting font families.
- Added text alignment styles

### CODE REVIEW

**Comment by 5anniversary** [comment]
https://github.com/ZhgChgLi/ZMarkupParser/pull/27#issuecomment-1672658065

**Comment by zhgchgli0718** [comment]
I apologize for being busy lately. I will review it before the end of August. Thank you for your contribution.

File: Sources/ZMarkupParser/HTML/Processor/HTMLTagAttributeToMarkupStyleVisitor.swift [hunk 1]
```diff
@@ -101,7 +101,7 @@ struct HTMLTagStyleAttributeToMarkupStyleVisitor: HTMLTagStyleAttributeVisitor {
     }
     
     func convert(fromPX string: String) -> Int? {
-        guard let regex = try? NSRegularExpression(pattern: "([0-9]+)p(x|t)"),
+        guard let regex = try? NSRegularExpression(pattern: "([0-9]+.?[0-9]*)p(x|t)"),
```
Comment (zhgchgli0718) [inline Sources/ZMarkupParser/HTML/Processor/HTMLTagAttributeToMarkupStyleVisitor.swift:104]: Q: Should we change Int to CGFloat as well?

Comment (5anniversary) [reply Sources/ZMarkupParser/HTML/Processor/HTMLTagAttributeToMarkupStyleVisitor.swift:104]: Oh thank you, I forgot to change it to cgfloat, my case was only for 12.0px so I didn't think of that case. 
I was getting it as 0px in the 12.0px case.

I'll change it.

File: Sources/ZMarkupParser/HTML/Processor/HTMLElementMarkupComponentMarkupStyleVisitor.swift [hunk 2]
```diff
@@ -160,8 +160,8 @@ extension HTMLElementMarkupComponentMarkupStyleVisitor {
                     continue
                 }
                 
-                let key = style[0].trimmingCharacters(in: .whitespacesAndNewlines)
-                let value = style[1].trimmingCharacters(in: .whitespacesAndNewlines)
+                let key = style[0].trimmingCharacters(in: CharacterSet(charactersIn: "' \n"))
```
[the remark this answers is not in the document]
Comment (5anniversary) [reply Sources/ZMarkupParser/HTML/Processor/HTMLElementMarkupComponentMarkupStyleVisitor.swift:163]: Sorry, I looked at the official Apple documentation again and whitespacesAndNewlines is correct. Thanks for the review, I'll take it back.

File: Tests/ZMarkupParserTests/HTML/AllTagStyleAttributesHasAddToBuilderDefaultListTests.swift [hunk 3]
```diff
@@ -63,4 +63,9 @@ private class StubVisitor: HTMLTagStyleAttributeVisitor {
         count += 1
         return count
     }
+
+    func visit(_ styleAttribute: TextAlignHTMLTagStyleAttribute) -> Int {
+        count += 1
+        return count
+    }
```
Comment (zhgchgli0718) [inline Tests/ZMarkupParserTests/HTML/AllTagStyleAttributesHasAddToBuilderDefaultListTests.swift:70]: Test Failed:
ZMarkupParserTests.AllTagStyleAttributesHasAddToBuilderDefaultListTests
testCheckDefaultList, XCTAssertEqual failed: ("8") is not equal to ("7") - Must added new pre-defined

should change `XCTAssertEqual(visitor.count, 7, "Must added new pre-defined HTMLTagStyleAttribute to ZHTMLParserBuilder.styleAttributes")` to
`XCTAssertEqual(visitor.count, 8, "Must added new pre-defined HTMLTagStyleAttribute to ZHTMLParserBuilder.styleAttributes")`

Comment (5anniversary) [reply Tests/ZMarkupParserTests/HTML/AllTagStyleAttributesHasAddToBuilderDefaultListTests.swift:70]: Confirm and change! db737b8

**Pull request merged**